### DISCUSSION TITLE
Thoughts about meshoptimizer with WebGPU

### DISCUSSION THREAD

**Discussion by Spiri0** [zeux/meshoptimizer#740] [Ideas]
Hi community, I am a passionate developer and work with three.js and WebGPU. I am thrilled by the enormous increase in performance that WebGPU enables.

Here is a WebGPU repo from me:
https://github.com/Spiri0/Threejs-WebGPU-IFFT-Ocean-V2

I use a lot of WebGPU compute shaders in it. The meshoptimizer makes me curious. Do I understand correctly that the entire model is loaded first?
If so, the compute shaders might be a very attractive performance option. I'm pretty familiar with WebGPU and three.js. 
Since you've all been working on the meshoptimizer project for a lot longer than me, I'm asking you what you think about the idea of ​​calculating the mesh optimization in the GPU instead of in the CPU?
Do my thoughts make sense or not?

**Comment by zeux** [comment]
meshoptimizer is designed to be used offline. The set of optimization algorithms is executed once, the resulting data is saved into some sort of runtime-friendly format, then it can be loaded on the user's device and rendered many times. Using GPUs makes little sense in that context. There is no plan for now to port any of the algorithms to GPU, whether using WebGPU or any other API; any cases like this would require very specialized algorithm designs anyhow.

**Comment by Spiri0** [comment]
Thank you for your detailed answer. I see I will have to deal with meshoptimizer a lot more to understand more
